components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/par/OAuth2ParEndpoint.java

```diff
@@ -144,6 +154,38 @@ private void handleValidation(HttpServletRequest request, MultivaluedMap<String,
 
     private boolean isRequestUriProvided(MultivaluedMap<String, String> params) {
 
-        return params.containsKey(OAuthConstants.OAuth20Params.REQUEST_URI);
+        return params.containsKey(OAuthConstants.OAuth20Params.REQUEST_URI) &&
+                !params.get(OAuthConstants.OAuth20Params.REQUEST_URI).isEmpty();
+    }
+
+    private void checkClientAuthentication(HttpServletRequest request) throws ParCoreException, ParClientException {
+        OAuthClientAuthnContext oAuthClientAuthnContext = getClientAuthnContext(request);
+        if (!oAuthClientAuthnContext.isAuthenticated()) {
```

Comment:
```suggestion
        if (oAuthClientAuthnContext.isAuthenticated()) {
           return;
        }
```
this would help reduce nested-ifs